tests/v2/validation/provisioning/rke2/etcd_backup_restore.go

```diff
@@ -0,0 +1,136 @@
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+)
+
+const (
+	localClusterName             = "local"
+	w1Name                       = "w1"
+	ingress1Name                 = "ingress1"
+	w1ServiceName                = "w1-svc"
+	w2Name                       = "w2"
+	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+)
+
+func createSnapshot(client *rancher.Client, clustername string, generation int, namespace string) error {
+	clusterObj, existingSteveAPIObj, err := getProvisioningClusterByName(client, clustername, namespace)
+	if err != nil {
+		return err
+	}
+
+	clusterObj.Spec.RKEConfig.ETCDSnapshotCreate = &rkev1.ETCDSnapshotCreate{
+		Generation: generation,
+	}
+
+	_, err = client.Steve.SteveType(clusters.ProvisioningSteveResouceType).Update(existingSteveAPIObj, clusterObj)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func restoreSnapshot(client *rancher.Client, clustername string, name string,
+	generation int, restoreconfig string, namespace string) error {
+
+	clusterObj, existingSteveAPIObj, err := getProvisioningClusterByName(client, clustername, namespace)
+	if err != nil {
+		return err
+	}
+
+	clusterObj.Spec.RKEConfig.ETCDSnapshotRestore = &rkev1.ETCDSnapshotRestore{
+		Name:             name,
+		Generation:       generation,
+		RestoreRKEConfig: restoreconfig,
+	}
+
+	_, err = client.Steve.SteveType(clusters.ProvisioningSteveResouceType).Update(existingSteveAPIObj, clusterObj)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getSnapshots(client *rancher.Client,
+	localClusterID string) ([]v1.SteveAPIObject, error) {
+
+	steveclient, err := client.Steve.ProxyDownstream(localClusterID)
+	if err != nil {
+		return nil, err
+	}
+	snapshotSteveObjList, err := steveclient.SteveType("rke.cattle.io.etcdsnapshot").List(&types.ListOpts{})
+	if err != nil {
+		return nil, err
+	}
+
+	return snapshotSteveObjList.Data, nil
+
+}
+
+func createRKE2NodeDriverCluster(client *rancher.Client, provider *Provider, clusterName string, k8sVersion string, namespace string, cni string) (*v1.SteveAPIObject, error) {
+
+	nodeRoles := []machinepools.NodeRoles{
+		{
+			ControlPlane: true,
+			Etcd:         false,
+			Worker:       false,
+			Quantity:     2,
+		},
+		{
+			ControlPlane: false,
+			Etcd:         true,
+			Worker:       false,
+			Quantity:     3,
+		},
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     3,
+		},
+	}
+
+	cloudCredential, err := provider.CloudCredFunc(client)
+	if err != nil {
+		return nil, err
+	}
+	generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
```

Comment:
@igomez06 we want to stray from `fmt` and more so use `logrus`, right? Or does it not matter here since this is not the actual test file?

Reply:
As it seems, this is not used for logging, `fmt.Sprintf` is used to generate a string with dynamic input.

Reply:
Yeah this is fine it isn't for logging. So no worries